src/stacks-hierarchy/Root_TicketAssistantStack/use-naviagtion-focus.ts

```diff
@@ -0,0 +1,25 @@
+import {useEffect, useRef} from 'react';
+import {giveFocus} from '@atb/utils/use-focus-on-load';
+import {CompositeNavigationProp} from '@react-navigation/native';
+
+interface UseNavigationFocusProps {
+  navigation: CompositeNavigationProp<any, any>;
+  screenName: string;
+}
+export const useNavigationFocus = ({
+  navigation,
+  screenName,
+}: UseNavigationFocusProps) => {
+  const focusRef = useRef(null);
+
+  useEffect(() => {
+    if (
+      navigation.getState().routeNames[navigation.getState().index] ==
```

Comment:
Whether the array lookup is a safe operation depends on the navigation state.
I'm not sure if it is always valid, and would add a safety check just for peace of mind ;)